cpp/src/datetime/datetime_ops.cu

```diff
@@ -76,9 +76,14 @@ struct extract_component_operator {
 
     if (time_since_midnight.count() < 0) { time_since_midnight += days(1); }
 
-    auto hrs_  = duration_cast<hours>(time_since_midnight);
-    auto mins_ = duration_cast<minutes>(time_since_midnight - hrs_);
-    auto secs_ = duration_cast<seconds>(time_since_midnight - hrs_ - mins_);
+    auto hrs_       = duration_cast<hours>(time_since_midnight);
+    auto mins_      = duration_cast<minutes>(time_since_midnight - hrs_);
+    auto secs_      = duration_cast<seconds>(time_since_midnight - hrs_ - mins_);
+    auto millisecs_ = duration_cast<milliseconds>(time_since_midnight - hrs_ - mins_ - secs_);
+    auto microsecs_ =
+      duration_cast<microseconds>(time_since_midnight - hrs_ - mins_ - secs_ - millisecs_);
+    auto nanoosecs_ = duration_cast<nanoseconds>(time_since_midnight - hrs_ - mins_ - secs_ -
```

Comment:
```suggestion
    auto nanosecs_ = duration_cast<nanoseconds>(time_since_midnight - hrs_ - mins_ - secs_ -
```

Reply:
Thanks for catching this David! I fixed the other usage too.